ecfr-parser/parseXML/parseXML.go

```diff
@@ -0,0 +1,323 @@
+package parseXML
+
+import (
+	"crypto/md5"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"strings"
+
+	"github.com/cmsgov/cmcs-eregulations/ecfr-parser/ecfr"
+)
+
+type PostProcesser interface {
+	PostProcess() error
+}
+
+func ParsePart(b io.ReadCloser) (*Part, error) {
+	d := xml.NewDecoder(b)
+
+	p := &Part{}
+	if err := d.Decode(p); err != nil {
+		return p, err
+	}
+	return p, nil
+}
+
+type Part struct {
+	XMLName   xml.Name        `xml:"DIV5" json:"-"`
+	Structure *ecfr.Structure `xml:"-" json:"structure"`
+	Citation  SectionCitation `xml:"N,attr" json:"label"`
+	Type      string          `xml:"TYPE,attr" json:"node_type"`
+	Header    string          `xml:"HEAD" json:"title"`
+	Authority string          `xml:"AUTH>PSPACE" json:"-"`
+	Source    string          `xml:"SOURCE>PSPACE" json:"-"`
+	Children  PartChildren    `xml:",any" json:"children"`
+}
+
+func (p *Part) PostProcess() (err error) {
+	for _, child := range p.Children {
+		c, ok := child.(PostProcesser)
+		if ok {
+			if err := c.PostProcess(); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+type PartChildren []interface{}
+
+func (c *PartChildren) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	switch start.Name.Local {
+	case "DIV6":
+		child := &Subpart{}
+		if err := d.DecodeElement(child, &start); err != nil {
+			return err
+		}
+		*c = append(*c, child)
+	case "DIV8":
+		child := &Section{}
+		if err := d.DecodeElement(child, &start); err != nil {
+			return err
+		}
+		*c = append(*c, child)
+	default:
+		log.Printf("[WARNING] Unknown XML type in Part: %+v\n", start)
+		d.Skip()
+	}
+
+	return nil
+}
```

Comment:
Might make sense to use `switch strings.ToLower(start.Name.Local)` to avoid the need for case-sensitive XML tags. For example, most seem uppercase but `img` is lowercase for some reason. Could cause problems later or if eCFR API changes and/or is inconsistent.

Reply:
It's good to allow these to be errors, the capitalization could be communicating meaning (until we know otherwise).